feat(sdk-generator): turn x-local-timezone mandatory for utils.Date (#1919)
AmadeusITGroup/otter: migration-guides/11.0.md

## @o3r/schematics

- `NodePackageNgAddTask` has been removed. `setupDependencies` should be used instead

> - The [`stringifyDate` cli option](https://www.npmjs.com/package/@ama-sdk/schematics#dates) is now enabled by default. If 

Reviewer: ```suggestion
- The [`stringifyDate` CLI option](https://www.npmjs.com/package/@ama-sdk/schematics#dates) is now enabled by default. If 
```

you want to use the `Date` format, you will need to set it to `false` via the command line:

```shell
# with yarn package manager
yarn run schematics @ama-sdk/schematics:typescript-core --global-property stringifyDate=false
# with npm package manager

> npm run schematics @ama-sdk/schematics:typescript-core --global-property stringifyDate=false

Reviewer 2: ```suggestion
npm exec schematics @ama-sdk/schematics:typescript-core -- --global-property stringifyDate=false
```

```

or by disabling it for your generator in your `openapitools.json` file:
```json5
{
  "$schema": "https://raw.githubusercontent.com/OpenAPITools/openapi-generator-cli/master/apps/generator-cli/src/config.schema.json",
  "generator-cli": {
    "version": "7.4.0", // Version of the Codegen downloaded by the cli - updated via the 
    "storageDir": ".openapi-generator",
    "generators": {
      "my-generator": {
        "generatorName": "typescriptFetch",
        "output": ".",
        "inputSpec": "spec-path.(yaml|json)",
        "globalProperty": {
          "stringifyDate": false
        }
      }
    }
  }
}
```

> - Prior version 11, the default type for dates was the `utils.Date` and `utils.DateTime` which take the date

Reviewer: ```suggestion
- Prior to version 11, the default type for dates was `utils.Date` and `utils.DateTime` which take the date
```

from the API response and remove the timezone to use the local one from the user's machine.

> If a date needed to be expressed in its original timezone, the date specification needed an extra `x-date-timezone` vendor.

Reviewer: ```suggestion
If a date had to be expressed in its original timezone, the date specification required an extra `x-date-timezone` vendor.
```

> From version 11, the default behavior is now to keep the timezone of the API response and to use `string` or `Date`

Reviewer: ```suggestion
As of version 11, the default behavior is now to keep the timezone of the API response and to use `string` or `Date`
```

depending on the `stringifyDate` value. The removal the timezone is now an 'opt-in' mechanism and dates with a timezone 

> that should be ignored needs to add the `x-local-timezone` vendor.

Reviewer: ```suggestion
that should be ignored need to add the `x-local-timezone` vendor.
```

> [!NOTE]

> > You can revert to the previous behaviour thanks to the global property `useLegacyDateExtension`

Reviewer 3: Not mandatory
```suggestion
> You can revert to the previous behavior thanks to the global property `useLegacyDateExtension`
```
